src/safeds_runner/server/_memoization_strategies.py

```diff
@@ -0,0 +1,54 @@
+"""Module that contains different memoization strategies."""
+
+from collections.abc import Callable
+from typing import Any, TypeAlias
+
+from safeds_runner.server._memoization_stats import MemoizationStats
+
+# Callable = Stat Key Extractor, Boolean = Reverse Order
+StatOrderExtractor: TypeAlias = tuple[Callable[[tuple[str, MemoizationStats]], Any], bool]
```

Comment:
The boolean to revert the order seems unnecessary. You can always negate the values in the strategies. Please, also add a short comment about the expected result of a strategy (values with low results get removed first IIUC).

Instead of `Any`, we could also force the return type to be a `float`. I can't think of a useful strategy where we don't return numbers.